maintenance/wikia/uncycloUsersMigrator.php

```diff
@@ -0,0 +1,695 @@
+	}
+
+	/**
+	 * Create a shared DB account using given Uncyclo account
+	 *
+	 * @param User $user
+	 * @return User|false created global account (with a new ID and migrated user settings)
+	 */
+	protected function doCreateGlobalUser( User $user ) {
+		// don't touch uncyclo users in "global users only" mode
+		if ( $this->onlyRenameGlobalUsers ) {
+			return;
+		}
+
+		$this->output( sprintf( "\n\tcreating a shared account for %s...", $user->getName() ) );
+
+		if ( $this->isDryRun ) {
+			return false;
+		}
+
+		/* @var User $extUser */
+		$extUser = clone $user;
+		$extUser->setToken();
+
+		// this code was borrowed from ExternalUser_Wikia::addToDatabase
+		$fname = __METHOD__;
+		$dbw = wfGetDB( DB_MASTER, [], self::SHARED_DB );
+		$dbw->begin( $fname );
+
+		try {
+			/**
+			 * Uncyclo does not have $wgSharedDB set so all queries from User class goes to the LOCAL database
+			 *
+			 * As we want to register and modify a GLOBAL user, set both $wgSharedDB and wgExternalSharedDB to "wikicities".
+			 * This wil make User::saveSettings write to the shared database
+			 */
+			$wrapper = new Wikia\Util\GlobalStateWrapper( [
+				'wgSharedDB'         => self::SHARED_DB,
+				'wgExternalSharedDB' => self::SHARED_DB,
+			] );
+			$wrapper->wrap(function () use ( $dbw, $extUser, $fname, $user ) {
+				$dbw->insert(
+					self::USER_TABLE,
+					[
+						'user_id' => null,
+						'user_name' => $user->getName(),
+						'user_real_name' => $user->getRealName(),
+						'user_password' => $user->mPassword,
+						'user_newpassword' => '',
+						'user_email' => $user->mEmail, // getEmail() would make a DB query
+						'user_touched' => '',
+						'user_token' => '',
+						'user_options' => '',
+						'user_registration' => $dbw->timestamp($user->mRegistration),
+						'user_editcount' => $user->getEditCount(), // use uncyclo counter
+						'user_birthdate' => $user->mBirthDate
+					],
+					$fname
+				);
+
+				$extUser->mId = $dbw->insertId();
+
+				// move user settings to the shared DB
+				$res = $this->getUncycloDB()->select(
+					'user_properties',
+					[ 'up_property', 'up_value' ],
+					array( 'up_user' => $user->getId() ),
+					$fname
+				);
+
+				foreach ( $res as $row ) {
+					$dbw->insert(
+						'user_properties',
+						[
+							'up_user'     => $extUser->mId,
+							'up_property' => $row->up_property,
+							'up_value'    => $row->up_value,
+						],
+						$fname
+					);
+				}
+
+				// mark migrated accounts
+				$dbw->insert(
+					'user_properties',
+					[
+						'up_user'     => $extUser->mId,
+						'up_property' => 'uncyclo_user',
+						'up_value'    => $user->getId(),
+					],
+					$fname
+				);
+			});
+
+			// we have a new ID for a global account
+			// update user ID in uncyclo database
+			$this->doChangeUncycloUserId( $user, $extUser->getId() );
+
+			// delete the user and his settings from the uncyclo DB
+			$this->deleteUncycloUser( $user );
```

Comment:
The solution for now is to pre-generate shared accounts for uncyclo accounts and update the `uncyclo.user` entries with the new shared ID.